commands/washtub/wash.js

```diff
@@ -1,30 +1,42 @@
 'use strict'
 
-let cli = require('heroku-cli-util')
 let co = require('co')
-let http = require('https')
 let wait = require('co-wait')
+let cli = require('heroku-cli-util')
+let http = require('https')
+let { sortBy, last, filter } = require('lodash')
 let { WashtubWash, load_wash } = require('../../lib/washtub')
 let { ensure_app, ensure_token } = require('../../lib/cli-util')
 
 function * run(context, heroku) {
-  let backupid = context.args.backup
-
-  if(! backupid.match(/^[a-z]+\d+$/)) {
-    throw new Error(`Please specify backup id.  ${backupid} does not seem to be a valid backup`)
-  }
-
-  let bid = backupid.replace(/^[a-z]+/, '')
+  let database = context.args.database || 'DATABASE_URL'
+  let database_basename = database.replace(/^(?:HEROKU_POSTGRESQL_)?(.*)(?:_URL)?$/, '$1')
   let app = ensure_app(context)
+  let target = context.args.target
 
   co(function *() {
-    cli.action.start("Performing database wash")
+    cli.action.start(`Performing database wash of ${database} into ${target}`)
     cli.action.status("looking up configs")
     let config = yield heroku.get(`/apps/${app}/config-vars`)
     cli.action.status('Looking up backup')
-    let path = `/client/v11/apps/${app}/transfers/${bid}?verbose=false`
+    let path = `/client/v11/apps/${app}/transfers?verbose=false`
     let options = { host: 'https://postgres-api.heroku.com' }
-    let backup = yield heroku.get(path, options)
+    let predicate = (o) => {
+      return o.from_name == database_basename && o.to_name == 'BACKUP'
+    }
+    let backups = filter(yield heroku.get(path, options), predicate)
+    let backup = last(sortBy(backups, ['num']))
+
+    console.log(backups)
+    console.log(backup)
```

Comment:
Do you want to remove these debug statements before we merge?

Reply:
Done.